mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -1913,6 +1912,128 @@ def _apply_order_by_search_logged_models(
+                identifier, op, value = non_whitespace_tokens
+                value = value.strip("'").strip('"')
+                if m := re.match(r"^([a-z]+)\.(.+)", identifier):
+                    comp = Comparison(
+                        entity=Entity.from_str(m.group(1)),
+                        key=m.group(2).strip("`"),
+                        op=op,
+                        value=value,
+                    )
+                else:
+                    comp = Comparison(
+                        entity=Entity.ATTRIBUTE,
+                        key=identifier.strip("`"),
+                        op=op,
+                        value=value,
+                    )
+                comparisons.append(comp)
+            # Ignore whitespace and 'AND' tokens, otherwise raise an error
+            elif (stripped := stmt.value.strip()) and stripped.lower() != "and":
+                raise MlflowException.invalid_parameter_value(
+                    f"Invalid filter string: {filter_string}. Expected a list of comparisons "
+                    f"separated by 'AND' (e.g. 'metrics.loss > 0.1 AND params.lr = 0.01')."
+                )
+
+        filters = []
+        dialect = self._get_dialect()
+        for comp in comparisons:
+            if comp.entity == Entity.ATTRIBUTE:
+                comp_func = SearchUtils.get_sql_comparison_func(comp.op, dialect)
+                key = SqlLoggedModel.ALIASES.get(comp.key, comp.key)
+                value = float(comp.value) if SqlLoggedModel.is_numeric(key) else comp.value
+                filters.append(comp_func(getattr(SqlLoggedModel, key), value))
+                continue
+
+            comp_func = SearchUtils.get_sql_comparison_func(comp.op, dialect)
+            if comp.entity == Entity.METRIC:
+                subquery = (
+                    session.query(SqlLoggedModelMetric)
+                    .filter(SqlLoggedModelMetric.metric_name == comp.key)
+                    .subquery()
+                )
+                filters.append(comp_func(SqlLoggedModelMetric.metric_value, float(comp.value)))
+            elif comp.entity == Entity.PARAM:
+                subquery = (
+                    session.query(SqlLoggedModelParam)
+                    .filter(SqlLoggedModelParam.param_key == comp.key)
+                    .subquery()
+                )
+                filters.append(comp_func(SqlLoggedModelParam.param_value, comp.value))
+            elif comp.entity == Entity.TAG:
+                subquery = (
+                    session.query(SqlLoggedModelTag)
+                    .filter(SqlLoggedModelTag.tag_key == comp.key)
+                    .subquery()
+                )
+                filters.append(comp_func(SqlLoggedModelTag.tag_value, comp.value))
+
+            models = models.join(subquery)
+
+        return models.filter(SqlLoggedModel.experiment_id.in_(experiment_ids), *filters)
+
     def search_logged_models(
         self,
         experiment_ids: list[str],
```

Comment:
Could we mention in the docstring about what's supported for filter_string? It's not clear to me right now